### ISSUE TITLE
Cannot create clusters using UI in Rancher 2.9.x

### ISSUE THREAD

**Issue by nnoethling** [nutanix/docker-machine#124]
In the current Rancher version 2.9.1 the Nutanix node driver is not working out of the box via UI.
When trying to create a new cluster with a network the UI show an validation Error and its not possible to create the cluster.

### Nutanix Cluster Information
Current versions

### Rancher Version
v2.9.1

### Rancher Node Driver Version
v3.6.0

### Node template
In Rancher version 2.8.x the vmNetwork field was an array field with multiple networks:
<img width="114" alt="image" src="https://github.com/user-attachments/assets/88df073f-e657-4d05-a31c-e52725ba94cc">

In 2.9.1 vmNetwork is not a field that can hold multiple networks
<img width="1430" alt="image" src="https://github.com/user-attachments/assets/859595d6-171b-4d63-92e5-8e0a6309651c">

### Debug Output
No logs visible in Rancher Pods regarding that issue.
UI Error message says: `NutanixConfig.rke-machine-config.cattle.io "nc-test-pool1-xwwj9" is invalid: vmNetwork: Invalid value: "string": vmNetwork in body must be of type array: "string"`
Even with empty Network field its not possible to create the Cluster.

Request response:
```
{
    "type": "error",
    "links": {},
    "code": "Invalid",
    "message": "NutanixConfig.rke-machine-config.cattle.io \"nc-test-pool1-vcggj\" is invalid: vmNetwork: Invalid value: \"string\": vmNetwork in body must be of type array: \"string\"",
    "status": 422,
    "type": "error"
}
```

Cluster Creation is working via helm chart or CRD.
I think this issue is based on a behaviour change in Rancher UI because vmNetwork and vmCategories are array values and are not displayed correctly as array field in Rancher.

### Expected Behavior
Cluster can be deployed via UI

### Actual Behavior
Cannot create cluster because Error is blocking creation with Error message `NutanixConfig.rke-machine-config.cattle.io "nc-test-pool1-xwwj9" is invalid: vmNetwork: Invalid value: "string": vmNetwork in body must be of type array: "string"`
​

**Comment by tuxtof** [comment]
Hello @nnoethling 

this is due to Rancher change between RKE1 and RKE2

with RKE2 you need to use the [nutanix rancher extension](https://github.com/nutanix-cloud-native/nutanix-rancher-extension) to correctly specific network 
with RKE1 it was part of the almost built-in UI

In all cases, for the moment, there are some issues with Rancher 2.9 and the external plugin that cause trouble with the entire process to deploy RKE2 cluster on Nutanix with the UI

we are actually working with Rancher, to find a fix

stay tuned

**Comment by Carl-ErikR** [comment]
Another issue with creating clusters using UI in Rancher 2.9.x https://github.com/rancher/rancher/issues/47663

**Comment by tuxtof** [comment]
fixed with Rancher 2.9.3
